packages/pwa-kit-extension-sdk/src/configs/webpack/overrides-resolver-loader.ts

```diff
@@ -161,67 +161,63 @@ const OverrideResolverLoader = function (this: LoaderContext<any>) {
         return false
     }
 
-    // Get full package path
-    const fullPackagePath = path.join(
-        projectDir,
-        isMonoRepo ? MONO_REPO_WORKSPACE_FOLDER : NODE_MODULES_FOLDER,
-        packagePath
-    )
+    let projectPath = ''
+    let isExtensionFile = false
+
+    // Check for extension in monorepo packages or node_modules
+    if (source.includes(`${path.sep}${MONO_REPO_WORKSPACE_FOLDER}${path.sep}`)) {
+        // Extract package path from monorepo structure
+        const [basePath, remainingPath] = source.split(
+            `${path.sep}${MONO_REPO_WORKSPACE_FOLDER}${path.sep}`
+        )
+        projectPath = `${basePath}${path.sep}${MONO_REPO_WORKSPACE_FOLDER}${path.sep}${
+            remainingPath.split(path.sep)[0]
+        }`
+    } else if (source.includes(`${path.sep}${NODE_MODULES_FOLDER}${path.sep}`)) {
+        // Special handling for scoped packages (@namespace/package)
+        const [basePath, packagePath] = source.split(`${path.sep}${NODE_MODULES_FOLDER}${path.sep}`)
+        const packageParts = packagePath.split(path.sep)
+
+        projectPath = packageParts[0].startsWith('@')
+            ? `${basePath}${path.sep}${NODE_MODULES_FOLDER}${path.sep}${packageParts[0]}${path.sep}${packageParts[1]}`
+            : `${basePath}${path.sep}${NODE_MODULES_FOLDER}${path.sep}${packageParts[0]}`
+    }
 
-    // Check if this package is an extension by looking for extension-meta.json
-    const isExtensionFile = isExtensionPackage(fullPackagePath)
+    isExtensionFile = isExtensionPackage(projectPath)
 
-    // Exit early if we have:
-    // 1. Processed this file already.
-    // 2. The file is not an extension file.
-    // 3. The file is an extension setup file.
-    if (targetCache.includes(source) || !isExtensionFile || isSetupFile) {
+    // Exit if it's not an extension file
+    if (!isExtensionFile) {
         return false
     }
 
-    // Because our webpack configuration is setup to resolve symlinks, we need to normalize the source path because
-    // the source path passed to the loaded is not representative of what you would see in a generated project (e.g.
-    // it doesn't resolve to being in the node_modules folder).
-    let normalizedSource = ''
-
-    // We are only concerned with the source path relative to the extension package namespace.
-    normalizedSource = `${
+    // Normalize the source path
+    // We are only concerned with the source path relative to the extension package namespace
+    const packagePath =
         source
             .split(
                 `${path.sep}${isMonoRepo ? MONO_REPO_WORKSPACE_FOLDER : NODE_MODULES_FOLDER}${
                     path.sep
                 }`
             )
             .pop() ?? ''
-    }`
 
     // At this point the path is either POSIX or windows, we need to normalize it to POSIX.
-    normalizedSource = normalizedSource.replace(/\\/g, '/')
+    let normalizedSource = packagePath.replace(/\\/g, '/')
```

Comment:
nit, based on your comment above, you might be able to use `path.posix.normalize()`? Not sure if that make any difference in our scenario